velox/common/memory/MmapAllocator.cpp

```diff
@@ -116,21 +123,34 @@ int64_t MmapAllocator::free(Allocation& allocation) {
   numAllocated_.fetch_sub(numFreed);
   return numFreed * kPageSize;
 }
-
 MachinePageCount MmapAllocator::freeInternal(Allocation& allocation) {
   if (allocation.numRuns() == 0) {
     return 0;
   }
   MachinePageCount numFreed = 0;
 
-  for (auto& sizeClass : sizeClasses_) {
-    numFreed += sizeClass->free(allocation);
+  for (auto i = 0; i < sizeClasses_.size(); ++i) {
+    auto& sizeClass = sizeClasses_[i];
+    int32_t pages = 0;
+    uint64_t clocks = 0;
+    {
+      ClockTimer timer(clocks);
+      pages = sizeClass->free(allocation);
+    }
+    if (pages) {
+      // Increment the free time only if the allocation contained
+      // pages in the class. Note that size class indices in the
```

Comment:
> Note that size class indices in the allocator are not necessarily the same as in the stats.

This is confusing. Any chance this can be designed differently?